lib/record_store/zone/config.rb

```diff
@@ -3,15 +3,17 @@ class Zone
     class Config
       include ActiveModel::Validations
 
-      attr_reader :ignore_patterns, :providers, :supports_alias
+      attr_reader :ignore_patterns, :providers, :supports_alias, :templates
 
       validate :validate_zone_config
 
-      def initialize(ignore_patterns: [], providers: nil, supports_alias: nil)
+      def initialize(ignore_patterns: [], providers: nil, supports_alias: nil, templates: [])
```

Comment:
I would perhaps have named this with a more explicit name to make sure that we know what those templates are for: `inject_records_templates` or `implicit_records_templates` for instance